### DISCUSSION TITLE
v2: SpongeChat[?], the second coming

### DISCUSSION THREAD

**Discussion by hyd3rs** [spongedsc/SpongeChat#67] [Announcements]
Welcome to v2! This release brings much-needed improvements, new features, and new uses of the Discord API itself. The bot's been remade from the ground up with a much-needed fresh coat of paint based off of the `create-discord-bot` boilerplate, a stable **multi-channel history**, slash commands, and much more!
 
 Below is a (non-exhaustive) list of what's new, and what's changed:
 
## What's new
- **Rewritten** based on `create-discord-bot` with a very flexible core, supports slash commands & actions (context menu) at launch
- **kV** [internal codename], a **Valkey/Dragonfly-compatible** key-value database system capable of fast reads/writes in-memory
- **Memories**, an efficient context window / conversation history system capable of per-channel conversations & logging
- **Imagine**, the direct successor to imageGen in v1; accessible through passive activation (talking in a channel where SpongeChat is listening) or through a command, `/imagine`
- **Silent mode**, to make it shut the f\*ck up when it's yapping too much
- **Instruction sets**, a quick way to change SpongeChat's personality in passive activation
- **Direct context windows** [to be renamed], the direct replacement for Readback; this can be found by right-clicking any SpongeChat-answered message and clicking "Direct context window"
 
Most, if not all features are present in 2.0. If we missed something, please open an issue and we'll try to get to it as soon as possible.
 
Cheers, everybody! 🥂

## Pull Requests
* [v2/feat] Image generation command by @daniwasonline in https://github.com/spongedsc/SpongeChat/pull/58
* [v2/refactor] Move interactionCreate to dedicated file/module, add Actions (context menu) support by @daniwasonline in https://github.com/spongedsc/SpongeChat/pull/59
* [v2/refactor] General chat improvements & code cleanup by @daniwasonline in https://github.com/spongedsc/SpongeChat/pull/60
* [v2/bug] Addressing context overloads by @daniwasonline in https://github.com/spongedsc/SpongeChat/pull/65
* [MAJOR] The v2 master-PR by @daniwasonline in https://github.com/spongedsc/SpongeChat/pull/55

## New Contributors
* @renovate made their first contribution in https://github.com/spongedsc/SpongeChat/pull/13
* @Butterroach made their first contribution in https://github.com/spongedsc/SpongeChat/pull/35
* @daniwasonline made their first contribution in https://github.com/spongedsc/SpongeChat/pull/54

**Full Changelog**: https://github.com/spongedsc/SpongeChat/commits/2.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/spongedsc/SpongeChat/releases/tag/2.0.0'>v2: SpongeChat[?], the second coming</a>.</em>

**Comment by Butterroach** [comment]
@renovate made their first contribution
